app/models/course/material/folder.rb

```diff
@@ -35,4 +55,14 @@ def validate_name_is_unique_among_materials
     conflicts = parent.materials.where('lower(name) = ?', name.downcase)
     errors.add(:name, :taken) if conflicts.any?
   end
+
+  def existing_names
+    parent.children.where.not(id: self).pluck(:name) + parent.materials.pluck(:name)
+  end
+
+  def assign_valid_name
+    return unless name_changed? && owner
```

Comment:
Do you have better suggestions ?  or you think that owner should handle this case it self ?